cli/sidecar/sidecar.go

```diff
@@ -146,6 +158,11 @@ func ConfigureSidecar(args []string) []string {
 		sidecarArgs = append(sidecarArgs, fmt.Sprintf("--cache_dir=%s", diskCacheDir))
 	}
 
+	if synchronousWriteFlag == "1" || synchronousWriteFlag == "true" {
+		sidecarArgs = append(sidecarArgs, "--synchronous_write")
+		args = append(args, "--bes_upload_mode=wait_for_upload_complete")
```

Comment:
I'm not sure this is enough for the BES to be sync because the sidecar fires and forgets the forwarded build events?

Reply:
For PublishBuildToolEventStream, we call stream.CloseSend() ](https://github.com/buildbuddy-io/buildbuddy/blob/93da3c661acd419d93a5bc329de216f220574c7d/server/build_event_protocol/build_event_server/build_event_server.go#L78) after we acked the everything. So if bazel is waiting for upload complete, it should be enough for Event Stream. 

But for PublishLifeCycleEvent, the server doesn't wait for the response from forwarded build events. I think in order to support the sync mode, I need to change the build_event_server to wait for the goroutines to finish before returning response behind a flag.

Reply:
ahh I see - For PublishBuildToolEventStream, the build_event_protocol server didn't check the response from the proxy.

Reply:
I updated the build_event_server to wait until forwarded clients acknowledged if sync mode is on.